fs/checksum_calculator.go

```diff
@@ -27,8 +34,30 @@ type calculatedFile struct {
 	err      error
 }
 
-// Sum returns a hex-encoded SHA256 checksum value of a file or directory given a path.
-func (c ChecksumCalculator) Sum(paths ...string) (string, error) {
+// SumAsHexString returns the sum of the underlying SHA256 hash in hex format.
+// Additional calls to the `Write(p []byte)` and `WritePath(paths ...strings)`
+// methods update the underlying SHA265 value of `ChecksumCalculator` so
+// `SumAsHexString()` can be called once or after any additional data
+// is written to ChecksumCalculator.
+func (c ChecksumCalculator) SumAsHexString() string {
+	return hex.EncodeToString(c.sumValue)
+}
+
+// Write a byte slice onto an internal SHA265 hash object so that it can be summed later.
+// This method can be called multiple times and interleaved with calls to
+// `WriteFiles(paths ...string)`
+func (c ChecksumCalculator) Write(p []byte) (int, error) {
+	n, err := c.hash.Write(p)
+	if err != nil {
+		return n, err
+	}
+	return n, err
```

Comment:
```suggestion
	return c.hash.Write(p)
```

Unless you intend to do something different in the error case, these are logically equivalent.

Reply:
Thanks! I'm debating if it makes sense to hide n entirely is all.